### PR TITLE
NIFI-3728: Detect 'truncate table' event, allow exclusion of schema change events in CaptureChangeMySQL

### PR DESCRIPTION
**Pull request by mattyb149**
Wouldn't be surprised if #1689 or this PR needs to be rebased, depending on which is merged first.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by ijokarumawak** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java [hunk 1]
```diff
@@ -263,6 +263,17 @@
             .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor INCLUDE_SCHEMA_CHANGES = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-include-schema-changes")
+            .displayName("Include Schema Change Events")
+            .description("Specifies whether to emit events corresponding to a schema change event (ALTER TABLE, e.g.) in the binary log. Set to true if the schema change events are "
+                    + "desired/necessary in the downstream flow, otherwise set to false, which suppresses generation of these events and can increase flow performance.")
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java]: Now that we're including things like truncate, I agree.  However we refer to "schema change" many times in the CDC code. I wrote up NIFI-3735 to cover the change in terminology across the CDC code. In the meantime however, I will just update this property to have the correct documentation.

Comment (ijokarumawak) [inline nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java]: Sounds good, thank you! Is this PR ready for another review cycle?

File: nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/test/groovy/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQLTest.groovy [hunk 2]
```diff
@@ -362,6 +363,79 @@ class CaptureChangeMySQLTest {
...
+        client.sendEvent(new Event(
+                [timestamp: new Date().time, eventType: EventType.XID, nextPosition: 40] as EventHeaderV4,
+                {} as EventData
+        ))
+
+        testRunner.run(1, true, false)
+
+        def resultFiles = testRunner.getFlowFilesForRelationship(CaptureChangeMySQL.REL_SUCCESS)
+        // No 'schema_change' events expected
+        List<String> expectedEventTypes = ([] + 'begin' + Collections.nCopies(3, 'insert') + 'commit')
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/test/groovy/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQLTest.groovy]: Done (set to true to include them again)

File: nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java (hunk 1 above)
Comment (mattyb149) [inline nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java]: Yes please and thank you!

**Comment by ijokarumawak** [comment]
All changes look good to me. Merging into master. Thanks @mattyb149 !

**Pull request closed without merging**